aztokenprovider/token_provider.go

```diff
@@ -127,7 +137,14 @@ func (provider *userTokenProvider) GetAccessToken(ctx context.Context, scopes []
 	}
 
 	currentUser, ok := azusercontext.GetCurrentUser(ctx)
-	if !ok {
+	if !ok || currentUser.IdToken == "" && !provider.usernameAssertion {
```

Comment:
Is this logic right? Should we still try to get the access token if currentUser is ok?

Reply:
I've reworked this!